diff --git a/node_modules/markdown-it-mentions/index.js b/node_modules/markdown-it-mentions/index.js
index a89b364..5dbdf41 100644
--- a/node_modules/markdown-it-mentions/index.js
+++ b/node_modules/markdown-it-mentions/index.js
@@ -22,4 +22,4 @@ function parser(match, utils) {
   ].join('');
 }
 
-module.exports = regexp(/@(\w+)/, parser);
+module.exports = regexp(/@([a-zA-Z0-9\-_.#$]{3,24})/, parser);
