diff --git a/node_modules/vite/dist/node/chunks/dep-1412e872.js b/node_modules/vite/dist/node/chunks/dep-1412e872.js
index 6d1b4a4..2ff2c0a 100644
--- a/node_modules/vite/dist/node/chunks/dep-1412e872.js
+++ b/node_modules/vite/dist/node/chunks/dep-1412e872.js
@@ -21705,10 +21705,6 @@ function resolveHtmlTransforms(plugins) {
     return [preHooks, postHooks];
 }
 async function applyHtmlTransforms(html, hooks, ctx) {
-    const headTags = [];
-    const headPrependTags = [];
-    const bodyTags = [];
-    const bodyPrependTags = [];
     for (const hook of hooks) {
         const res = await hook(html, ctx);
         if (!res) {
@@ -21726,6 +21722,10 @@ async function applyHtmlTransforms(html, hooks, ctx) {
                 html = res.html || html;
                 tags = res.tags;
             }
+            const headTags = [];
+            const headPrependTags = [];
+            const bodyTags = [];
+            const bodyPrependTags = [];
             for (const tag of tags) {
                 if (tag.injectTo === 'body') {
                     bodyTags.push(tag);
@@ -21740,21 +21740,20 @@ async function applyHtmlTransforms(html, hooks, ctx) {
                     headPrependTags.push(tag);
                 }
             }
+            if (headPrependTags.length) {
+                html = injectToHead(html, headPrependTags, true);
+            }
+            if (headTags.length) {
+                html = injectToHead(html, headTags);
+            }
+            if (bodyPrependTags.length) {
+                html = injectToBody(html, bodyPrependTags, true);
+            }
+            if (bodyTags.length) {
+                html = injectToBody(html, bodyTags);
+            }
         }
     }
-    // inject tags
-    if (headPrependTags.length) {
-        html = injectToHead(html, headPrependTags, true);
-    }
-    if (headTags.length) {
-        html = injectToHead(html, headTags);
-    }
-    if (bodyPrependTags.length) {
-        html = injectToBody(html, bodyPrependTags, true);
-    }
-    if (bodyTags.length) {
-        html = injectToBody(html, bodyTags);
-    }
     return html;
 }
 const importRE = /\bimport\s*("[^"]*[^\\]"|'[^']*[^\\]');*/g;
